contracts/simple_airdrop/src/contract.rs

```diff
@@ -0,0 +1,387 @@
+            return Err(StdError::generic_err(
+                "from_timestamp can't be changed after window starts",
+            ));
+        }
+        config.from_timestamp = from_timestamp;
+        attributes.push(attr("new_from_timestamp", from_timestamp.to_string()))
+    }
+
+    if let Some(to_timestamp) = to_timestamp {
+        if env.block.time.seconds() >= config.from_timestamp && to_timestamp < config.to_timestamp {
+            return Err(StdError::generic_err(
+                "When window starts to_timestamp can only be increased",
+            ));
+        }
+        config.to_timestamp = to_timestamp;
+        attributes.push(attr("new_to_timestamp", to_timestamp.to_string()))
+    }
+
+    if config.to_timestamp <= config.from_timestamp {
+        return Err(StdError::generic_err("Invalid airdrop claim window"));
+    }
+
+    CONFIG.save(deps.storage, &config)?;
+    Ok(Response::new().add_attributes(attributes))
+}
+
+/// @dev Facilitates increasing ASTRO airdrop amount
+pub fn handle_increase_astro_incentives(
+    deps: DepsMut,
+    amount: Uint128,
+) -> Result<Response, StdError> {
+    let mut state = STATE.load(deps.storage)?;
+    state.total_airdrop_size += amount;
+    state.unclaimed_tokens += amount;
+
+    STATE.save(deps.storage, &state)?;
+    Ok(Response::new()
+        .add_attribute("action", "astro_airdrop_increased")
+        .add_attribute("total_airdrop_size", state.total_airdrop_size))
+}
+
+/// @dev Executes an airdrop claim for a Terra User
+/// @param claim_amount : Airdrop to be claimed by the user
+/// @param merkle_proof : Array of hashes to prove the input is a leaf of the Merkle Tree
+/// @param root_index : Merkle Tree root identifier to be used for verification
+pub fn handle_claim(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    claim_amount: Uint128,
+    merkle_proof: Vec<String>,
+    root_index: u32,
+) -> Result<Response, StdError> {
+    let recipient = info.sender;
+
+    let config = CONFIG.load(deps.storage)?;
+    let mut state = STATE.load(deps.storage)?;
+
+    // CHECK :: IS AIRDROP CLAIM WINDOW OPEN ?
+    if config.from_timestamp > env.block.time.seconds() {
+        return Err(StdError::generic_err("Claim not allowed"));
+    }
+
+    // CHECK :: IS AIRDROP CLAIM WINDOW OPEN ?
+    if config.to_timestamp < env.block.time.seconds() {
+        return Err(StdError::generic_err("Claim period has concluded"));
+    }
+
+    let merkle_root = config.merkle_roots.get(root_index as usize);
+    if merkle_root.is_none() {
+        return Err(StdError::generic_err("Incorrect Merkle Root Index"));
+    }
+
+    if !verify_claim(&recipient, claim_amount, merkle_proof, merkle_root.unwrap()) {
+        return Err(StdError::generic_err("Incorrect Merkle Proof"));
+    }
+
+    let mut user_info = USERS.load(deps.storage, &recipient).unwrap_or_default();
+
+    // Check if addr has already claimed the tokens
+    if !user_info.airdrop_amount.is_zero() {
```

Comment:
Is user available to claim the airdrop only once per address?

Reply:
yes, only once per address (or do we want to remove this once per address limit ?)

Reply:
no, it's fine